storage/archive/ha_archive.cc

```diff
@@ -1556,7 +1556,7 @@ int ha_archive::optimize(THD* thd, HA_CHECK_OPT* check_opt)
   if (init_archive_reader())
   {
     mysql_mutex_unlock(&share->mutex);
-    DBUG_RETURN(errno);
+    DBUG_RETURN(HA_ERR_CRASHED_ON_USAGE);
```

Comment:
Please don't change errno to HA_ERR_CRASHED_ON_USAGE.
This will hide the original error !